compiler/qsc/src/interpret/stateless.rs

```diff
@@ -152,11 +152,14 @@ impl Interpreter {
     }
 
     #[must_use]
-    pub fn new_eval_context(&self) -> EvalContext {
+    pub fn new_eval_context<S: Backend<ResultType = impl Into<val::Result>>>(
+        &self,
+        sim: S,
```

Comment:
Why make this an argument if all the call sites pass in `SparseSim::new()` now?

Reply:
It used to be all but one, but after a refactor I see you are right, all remaining callers pass the sparse simulator. I think I can put this back then, and we can use the stateful interpreter as a way to generate QIR from contexts like Python. Good catch, not sure how I missed that!